DelvEdit/src/com/interrupt/dungeoneer/editor/ui/menu/generator/LevelGeneratorMenuItem.java

```diff
@@ -0,0 +1,118 @@
+                needsRefresh = false;
+
+                if (menuItem.subMenu != null) {
+                    menuItem.subMenu.items.clear();
+                }
+
+                Array<LevelGeneratorInfo> levelGenerators = Editor.app.generatorInfo.getLevelGenerators();
+
+                if (levelGenerators.size > 0) {
+                    for (LevelGeneratorInfo generator : levelGenerators) {
+                        MenuItem parent = new MenuItem(generator.name.toUpperCase(), skin);
+
+                        if (generator.templates.size > 0) {
+                            for (LevelTemplateInfo template : generator.templates) {
+                                String label = template.roomGeneratorType + " (" + template.theme + ")";
+                                parent.addItem(new MenuItem(label, skin,
+                                        makeLevelGeneratorAction(template.theme, template.roomGeneratorType)));
+                            }
+                        } else {
+                            MenuItem item = new MenuItem("No templates found", skin);
+                            item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                            parent.addItem(item);
+                        }
+
+                        menuItem.addItem(parent);
+                    }
+
+                    menuItem.addSeparator();
+                    String label = "Re-Generate Level" + (Editor.app.generatorInfo.lastGeneratedLevelType != null
+                            ? " (" + Editor.app.generatorInfo.lastGeneratedLevelType + ", " + Editor.app.generatorInfo.lastGeneratedLevelRoomType + ")"
+                            : "");
+                    MenuItem item = new MenuItem(label, skin, makeAnotherLevelGeneratorAction())
+                            .setAccelerator(new MenuAccelerator(Keys.L, false, true));
+
+                    if (Editor.app.generatorInfo.lastGeneratedLevelType == null) {
+                        item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                        // TODO: Change color of accelerator label.
+                    }
+
+                    menuItem.addItem(item);
+                } else {
+                    MenuItem item = new MenuItem("No levels found", skin);
+                    item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                    menuItem.addItem(item);
+                }
+            }
+
+            private ActionListener makeLevelGeneratorAction(String theme, String roomGenerator) {
+                return new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        if (theme != null || roomGenerator != null) {
+                            Level level = Editor.app.getLevel();
+
+                            if (level != null) {
+                                level.editorMarkers.clear();
+                                level.entities.clear();
```

Comment:
I believe we should also be clearing a few other lists here too now:
```
Editor.app.getLevel().non_collidable_entities.clear();
Editor.app.getLevel().static_entities.clear();
```

Reply:
Fixed. That should get rid of #115.